microprofile/WFLY-14940_upgrade-microprofile-health-3.1.adoc

```diff
@@ -0,0 +1,69 @@
+= Upgrade MicroProfile Health to 3.1
+:author:            Martin Stefanko
+:email:             mstefank@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          microprofile,health,observability
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+
+MicroProfile 4.1 provides a minor upgrade for https://github.com/eclipse/microprofile-health/releases/tag/3.1[MicroProfile Health 3.1] with changes described in its https://download.eclipse.org/microprofile/microprofile-health-3.1/microprofile-health-spec-3.1.html#release_notes_3_1[Release Notes].
+
+This component upgrade also covers the upgrade of smallrye-health 3.1.1 that implements MicroProfile Health 3.1.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-14940[WFLY-14040]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1763[EAP7-1763]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+=== Testing By
+
+[X] Engineering
+[ ] QE
+
+== Requirements
+
+* Upgrade MicroProfile Health to 3.1
+* Upgrade SmallRye Health to 3.1.1
+* Pass all MicroProfile Health tests in WildFly integration test suite
+* Pass the MicroProfile Health 3.1 TCK without failures
+* Adjust the tests to work as needed
+* Adjust QuickStarts to showcase new Startup feature
```

Comment:
From this we get that the Quickstarts project is affected, too. So we should add it to the list above.

Reply:
BTW - the issue for updating the Quickstarts (if any but it should exist and be independent maybe?) should be added as well to the list of `Related issues`. 
WDYT?

Reply:
@fabiobrz  it depends whether we want to showcase `@Startup` in Health QS. There is no issue yet. @emmartins WDYT?

Reply:
We could also keep the AD as it is now. The process allows for this potential change (i.e. mentioning the Quickstarts here) later if we need to.
For sure it'd be nice to know whether there's a plan for that. WDYT @emmartins ?

Reply:
I've created https://issues.redhat.com/browse/WFLY-15089 and linked it here.

Reply:
Nice, thanks!